content/en/docs/Responses/ArtistWithAlbumsID3.md

```diff
@@ -0,0 +1,108 @@
+---
+
+{{< tabpane persistLang=false >}}
+{{< tab header="**Example**:" disabled=true />}}
+{{< tab header="OpenSubsonic" lang="json">}}
+{
+  "id": "37ec820ca7193e17040c98f7da7c4b51",
+  "name": "2 Mello",
+  "coverArt": "ar-37ec820ca7193e17040c98f7da7c4b51_0",
+  "albumCount": 1,
+  "userRating": 5,
+  "artistImageUrl": "https://demo.org/image.jpg",
+  "starred": "2017-04-11T10:42:50.842Z",
+  "musicBrainzId": "189002e7-3285-4e2e-92a3-7f6c30d407a2",
+  "sortName": "Mello (2)",
+  "roles": [
+    "artist",
+    "albumartist",
+    "composer"
+  ],
+  "album": [
+    {
+      "id": "200000002",
+      "parent": "100000002",
+      "album": "Colorsmoke EP",
+      "title": "Colorsmoke EP",
+      "name": "Colorsmoke EP",
+      "isDir": true,
+      "coverArt": "al-200000002",
+      "songCount": 12,
+      "created": "2021-02-23T04:24:48+00:00",
+      "duration": 4568,
+      "playCount": 1,
+      "artistId": "100000002",
+      "artist": "Synthetic",
+      "year": 2007,
+      "genre": "Electronic",
+      "userRating": 5,
+      "averageRating": 3,
+      "starred": "2021-02-22T05:51:53Z"
+    }
+  ]
+}
+{{< /tab >}}
+{{< tab header="Subsonic" lang="json" >}}
+{
+  "id": "37ec820ca7193e17040c98f7da7c4b51",
+  "name": "2 Mello",
+  "coverArt": "ar-37ec820ca7193e17040c98f7da7c4b51_0",
+  "albumCount": 1,
+  "userRating": 5,
+  "artistImageUrl": "https://demo.org/image.jpg",
+  "starred": "2017-04-11T10:42:50.842Z",
+  "album": [
+    {
+      "id": "200000002",
+      "parent": "100000002",
+      "album": "Colorsmoke EP",
+      "title": "Colorsmoke EP",
+      "name": "Colorsmoke EP",
+      "isDir": true,
+      "coverArt": "al-200000002",
+      "songCount": 12,
+      "created": "2021-02-23T04:24:48+00:00",
+      "duration": 4568,
+      "playCount": 1,
+      "artistId": "100000002",
+      "artist": "Synthetic",
+      "year": 2007,
+      "genre": "Electronic",
+      "userRating": 5,
+      "averageRating": 3,
+      "starred": "2021-02-22T05:51:53Z"
+    }
+  ]
+}
+{{< /tab >}}
+{{< /tabpane >}}
+
+| Field            | Type                               | Req.    | OpenS.  | Details                                                |
+| ---------------- | ---------------------------------- | ------- | ------- | ------------------------------------------------------ |
+| `id`             | `string`                           | **Yes** |         | The id of the arist                                    |
+| `name`           | `string`                           | **Yes** |         | The artist name.                                       |
+| `coverArt`       | `string`                           | No      |         | A covertArt id.                                        |
+| `artistImageUrl` | `string`                           | No      |         | An url to an external image source.                    |
+| `albumCount`     | `int`                              | **Yes** |         | Artist album count.                                    |
```

Comment:
It would be great to specify wether it is the number of albums this artist appears on or just the album this artist is credited as album artist

Reply:
That's a good question. Unfortunately I don't know how OG subsonic behaves.

Reply:
IIRC, in OG subsonic `getArtist` only handles AlbumArtists, so this would be the number of albums this artist is credited as album artist.

Reply:
Maybe we could just say it has to be consistent with what the server returns when getting all the albums for this artist?
Because now with OS we likely want all albums (including albums where the artist "appears on") and this would be backward compatible.
Furthermore, poorly tagged albums don't have albumartist tags and servers usually fallback on artist in that case.
What do you think?

Reply:
> Because now with OS we likely want all albums (including albums where the artist "appears on") and this would be backward compatible.

I'm not sure we should do this, as the client won't have a way to figure out what is the role of the artist in each album, as the AlbumID3 does not include the `contributors` attribute. Maybe we should add it?

Reply:
I would also like to note that `getIndexes` (and `search2`) both return an `artist` (without album count). I get these are legacy endpoints, but I think it would might make more sense to only do `albumArtist`. Perhaps a new `getArtist2` endpoint that accepts a list of roles instead?

One other note, if this specific discussion goes on (or dies) without this specific question resolved, I don't think it should hold back this PR (as the point is to fix other endpoints).

Reply:
> > Because now with OS we likely want all albums (including albums where the artist "appears on") and this would be backward compatible.
> 
> I'm not sure we should do this, as the client won't have a way to figure out what is the role of the artist in each album, as the AlbumID3 does not include the `contributors` attribute. Maybe we should add it?

Ah you are right, and not sure for contributors: I remember this pr https://github.com/opensubsonic/open-subsonic-api/pull/130 that was not approved. So indeed let's focus on one problem at a time